lib/random.gd

```diff
@@ -69,9 +69,86 @@ DeclareCategory( "IsRandomSource", IsComponentObjectRep );
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
-DeclareOperation( "Random", [IsRandomSource, IsList] );
+DeclareOperation( "Random", [IsRandomSource, IsListOrCollection] );
 DeclareOperation( "Random", [IsRandomSource, IsInt, IsInt] );
 
+##############################################################################
+##
+##  <#GAPDoc Label="InstallMethodWithGlobalRandomGen">
+##  <ManSection>
+##  <Func Name="InstallMethodWithGlobalRandomGen"
+##   Arg="opr,info[,famp],args-filts[,val],method"/>
+##
+##  <Description>
+##  This function is designed to simplify adding new overloads for
+##  <Ref Oper="Random"/> and <Ref Oper="PseudoRandom"/> to GAP which can
+##  be called both with, and without, a random soure.
+##  <P/>
+##  It accepts the same arguments as <Ref Func="InstallMethod"/>, with
+##  the extra requirement that the first member of <A>args-filts</A> must
+##  be IsRandomSource, and the <A>info</A> argument
+##  is compulsory and must begin 'for a random source and'.
+##  <P/>
+##  This function calls <Ref Func="InstallMethod"/> twice. Firstly
+##  with it's arguments unchanged, and secondly with the first argument
+##  removed, instead always passing in the value
+##  <Ref Var="GlobalRandomSource"/>.
```

Comment:
Hrm... that only "almost" explains what's going on. I think we should either  give a high-level description, or a full explanation, or perhaps both. How about this:

"This function then installs two methods: first it calls InstallMethod with unchanged arguments. Then it calls InstallMethod a second time to install anothr method which lacks the initial random source argument; this additional method simply invokes the original method, with GlobalRandomSource added as first argument."